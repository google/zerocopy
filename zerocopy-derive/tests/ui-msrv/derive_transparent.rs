../ui-nightly/derive_transparent.rs