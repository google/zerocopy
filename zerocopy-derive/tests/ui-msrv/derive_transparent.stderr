error[E0277]: the trait bound `NotZerocopy: FromZeroes` is not satisfied
  --> tests/ui-msrv/derive_transparent.rs:37:1
   |
37 | assert_impl_all!(TransparentStruct<NotZerocopy>: FromZeroes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromZeroes` is not implemented for `NotZerocopy`
   |
note: required because of the requirements on the impl of `FromZeroes` for `TransparentStruct<NotZerocopy>`
  --> tests/ui-msrv/derive_transparent.rs:27:19
   |
27 | #[derive(AsBytes, FromZeroes, FromBytes, Unaligned)]
   |                   ^^^^^^^^^^
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui-msrv/derive_transparent.rs:37:1
   |
37 | assert_impl_all!(TransparentStruct<NotZerocopy>: FromZeroes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
   = note: this error originates in the macro `assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotZerocopy: FromBytes` is not satisfied
  --> tests/ui-msrv/derive_transparent.rs:38:1
   |
38 | assert_impl_all!(TransparentStruct<NotZerocopy>: FromBytes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromBytes` is not implemented for `NotZerocopy`
   |
note: required because of the requirements on the impl of `FromBytes` for `TransparentStruct<NotZerocopy>`
  --> tests/ui-msrv/derive_transparent.rs:27:31
   |
27 | #[derive(AsBytes, FromZeroes, FromBytes, Unaligned)]
   |                               ^^^^^^^^^
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui-msrv/derive_transparent.rs:38:1
   |
38 | assert_impl_all!(TransparentStruct<NotZerocopy>: FromBytes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
   = note: this error originates in the macro `assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotZerocopy: AsBytes` is not satisfied
  --> tests/ui-msrv/derive_transparent.rs:39:1
   |
39 | assert_impl_all!(TransparentStruct<NotZerocopy>: AsBytes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `AsBytes` is not implemented for `NotZerocopy`
   |
note: required because of the requirements on the impl of `AsBytes` for `TransparentStruct<NotZerocopy>`
  --> tests/ui-msrv/derive_transparent.rs:27:10
   |
27 | #[derive(AsBytes, FromZeroes, FromBytes, Unaligned)]
   |          ^^^^^^^
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui-msrv/derive_transparent.rs:39:1
   |
39 | assert_impl_all!(TransparentStruct<NotZerocopy>: AsBytes);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
   = note: this error originates in the macro `assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `NotZerocopy: Unaligned` is not satisfied
  --> tests/ui-msrv/derive_transparent.rs:40:1
   |
40 | assert_impl_all!(TransparentStruct<NotZerocopy>: Unaligned);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Unaligned` is not implemented for `NotZerocopy`
   |
note: required because of the requirements on the impl of `Unaligned` for `TransparentStruct<NotZerocopy>`
  --> tests/ui-msrv/derive_transparent.rs:27:42
   |
27 | #[derive(AsBytes, FromZeroes, FromBytes, Unaligned)]
   |                                          ^^^^^^^^^
note: required by a bound in `_::{closure#0}::assert_impl_all`
  --> tests/ui-msrv/derive_transparent.rs:40:1
   |
40 | assert_impl_all!(TransparentStruct<NotZerocopy>: Unaligned);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `_::{closure#0}::assert_impl_all`
   = note: this error originates in the macro `assert_impl_all` (in Nightly builds, run with -Z macro-backtrace for more info)
